lib/nexosis_api/paged_array.rb

```diff
@@ -0,0 +1,43 @@
+module NexosisApi
+  # Generic list base class for list responses
+  # @since 1.4
+  class PagedArray < Array
+    def initialize(paged_response, item_array = [])
+      self[0..item_array.length] = item_array
+      paged_response.each do |k, v|
```

Comment:
You could probably use instance_variable_set to do this and eliminate a lot of the if/else syntax. https://ruby-doc.org/core-2.4.2/Object.html#method-i-instance_variable_set

Reply:
```
var_map = { 'pageNumber' => '@page_number', 'totalPages' => '@total_pages', ... }
paged_response.each { |k,v| instance_variable_set(var_map[k], v) }
```

Obviously this doesn't work for links.